code/socialDistribution/views.py

```diff
@@ -378,7 +378,7 @@ def editPost(request, id):
         return HttpResponseBadRequest("Only public posts are editable")
 
     if request.method == 'POST':
-        form = PostForm(request.POST, request.FILES, user=author)
+        form = PostForm(request.POST, request.FILES, user_id=author.id)
```

Comment:
we have to remove .id from the end of line 375. Otherwise, this will throw an error on post edit